src/functions.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-use-before-define */
+/* eslint-disable brace-style */
+// select variables
+const todoListElement = document.getElementById('todo');
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+const todoInput = document.querySelector('input');
+/* eslint-disable no-unused-vars */
+let EditTodoId = -1;
+// first render
+
+document.querySelector('#add-div').addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    // code for enter
+    saveTodo();
+    clearInputField();
+    renderTodos();
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }
+});
+
+// save Todo function
+/* eslint-disable consistent-return */
+export function saveTodo() {
+  const todoValue = todoInput.value;
+  const isEmpty = todoValue === '';
+  // check duplicate input
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+  // check emptyinput
+  if (isEmpty) {
+    return null;
+    /* eslint-disable no-else-return */
+  } else if (isDuplicate) {
+    return null;
+  }
+  //   else {
+  //             if (EditTodoId => 0) {
+  // todos = todos.map((todo, index) => ({
+  //             ...todo,
+  /* eslint-disable no-trailing-spaces */
+  //             value: index === EditTodoId ? todoValue: todo.value,
+  //         }))
+  //         EditTodoId = -1;
+  //   }     // update the edittodo
+  else {
+    todos.push({
+      value: todoValue,
+      completed: false,
+      index: todos.length,
```

Comment:
- [ ] As mentioned in the project requirements, the index of the todo item should be it's position; meaning the 1st item should have index 1 not 0. Kindly update this line to `index: todos.length + 1` to fix this issue 👍 

![image](https://user-images.githubusercontent.com/50721479/206715209-0ed039e6-ce46-4dca-9e58-02b20caf376c.png)